CandyMarket/CandyMarket/DataAccess/CandyRepo.cs

```diff
@@ -181,6 +197,33 @@ from UserCandies
             }
         }
 
+        // ConsumeRandomCandy: finds UserCandyId for oldest piece of a random candy by flavor and calls EatCandy
+        public UserCandyDetailed ConsumeRandomCandy(string flavor, int userId)
+        {
+            var sql = @"select top 1 with ties UserCandyId
+                        from UserCandies uc
+                            join Candies c
+                            on uc.candyId = c.candyId
+                        where isConsumed = 0
+                        and flavorCategory = @Flavor
+                        and userId = @UserId
+                        order by dateReceived";
+
+            using (var db = new SqlConnection(connectionString))
+            {
+                var parameters = new
+                {
+                    Flavor = flavor,
```

Comment:
This will need to be FlavorCategory instead of Flavor....so FlavorCategory = flavor; the property must match the view so Dapper can map it to the right things in the query.

Reply:
updated variable names flavor/Flavor to flavorCategory/FlavorCategory